tools/cpp/windows_cc_configure.bzl

```diff
@@ -93,6 +93,20 @@ def find_vc_path(repository_ctx):
     _auto_configure_warning_maybe(repository_ctx, "'BAZEL_VC' is not set, " +
                                                   "start looking for the latest Visual C++ installed.")
 
+    if "BAZEL_VC_TOOL" in repository_ctx.os.environ:
+        auto_configure_warning("'BAZEL_VC_TOOL' is set, " +
+                               "but 'BAZEL_VC' and 'BAZEL_VS' are not. " + 
+                               "This is not recomended and can " +
+                               "lead to incorrect VC++ detection and configuration.")
+        vc_tool_path = repository_ctx.os.environ["BAZEL_VC_TOOL"]
+        # As neither 'BAZEL_VC' nor 'BAZEL_VS' are set, we assume BAZEL_VC_TOOL looks like that: %VS_PATH%\VC\<path_to_tool>
+        # path_to_tool is different for different VS versions, but we don't care
+        # We DO NOT search in any other places as we can easily 
+        # get inconsistancy with VC++ tools and VCVARSALL.BAT 
+        # (assuming multiple versions of VS are installed)
+        vc_path = repository_ctx.re.search(r".*(^|[/\\])VC[\\/]?", vc_tool_path)
```

Comment:
`^` means line start. Is this regex correct?

Also, `repository_ctx` has no "re" member: https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html

Reply:
`^` allows to handle lines which starts from 'VC'. 
For example in line`VC/relative/path/` regex will math 'VC/'
I don't think if relative paths are allowed here, but it will not break anything anyway.

Reply:
Example:

```python
def _vc_path(p):
  p = p.replace("\\", "/").rstrip("/")
  return p[:-3] if p.endswith("/VC") else ""


vc_path = _vc_path(vc_tool_path)
```

Reply:
My first though was to use python inside `repository_ctx.execute `, but as it will make that code depend on python I just replaced with very naive equivalent algorithm.